src/react-components/presence-log.js

```diff
@@ -1,40 +1,40 @@
-import React, { Component } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import PropTypes from "prop-types";
 import styles from "../assets/stylesheets/presence-log.scss";
 import classNames from "classnames";
-import { injectIntl } from "react-intl";
 import { formatSystemMessage } from "./room/ChatSidebar";
 
 import ChatMessage from "./chat-message";
 import PhotoMessage from "./photo-message";
 import VideoMessage from "./video-message";
 import ImageMessage from "./image-message";
 import { getPresenceContextForSession } from "../utils/phoenix-utils";
+import { useIntl } from "react-intl";
+import PermissionMessage from "./permission-message";
+import { useCan } from "./room/useCan";
 
-class PresenceLog extends Component {
-  static propTypes = {
-    entries: PropTypes.array,
-    inRoom: PropTypes.bool,
-    hubId: PropTypes.string,
-    history: PropTypes.object,
-    presences: PropTypes.object,
-    onViewProfile: PropTypes.func,
-    intl: PropTypes.object
-  };
+export const presets = [
+  "InRoom",
+  "Notifications"
+];
 
-  constructor(props) {
-    super(props);
-  }
+export function PresenceLog({ entries, preset, hubId, history, presences, onViewProfile, include, exclude, ...rest }) {
+  const intl = useIntl();
+  const isMod = useCan("kick_users");
```

Comment:
We seem to be mixing moderator and owner in a few places. Here we are checking of `kick_user` permission as "mod" but in other spots related to this we are checking if they have the `owner` role. Is that intended?

Reply:
I think in our case we want the room owners to be the ones with voice/chat permissions so I've introduced a `useRole` hook to check for owner permissions and add some consistency.

Reply:
Should this be using `useRole` then?

Reply:
I did that in https://github.com/mozilla/hubs/pull/5731/commits/960262572ee32e244d7bb07b672e504cfa6a061b maybe you are referring to some other one. I think I've moved all of them.

Reply:
ah must have been looking at old code.